File: 345/solution.rs
impl Solution {
    pub fn reverse_vowels(s: String) -> String {
        let possible_vowels = "aeiouAEIOU";
        let mut result = String::new();
        let mut vowels = String::new();

        for c in s.chars() {
            if possible_vowels.contains(c) {
                vowels.push(c);
            }
        }

        for c in s.chars() {
            if possible_vowels.contains(c) {
                result.push(vowels.pop().unwrap());
            } else {
                result.push(c);
            }
        }

        result
    }
}

File: 345/solution.py
VOWELS = set("aeiouAEIOU")


class Solution:
    def reverse_vowels(self, s: str) -> str:
        def is_vowel(c: str) -> bool:
            return c in VOWELS

        vowels = [c for c in s if is_vowel(c)]

        return "".join(vowels.pop() if is_vowel(c) else c for c in s)
